server/app/services/applications/ApplicationService.java

```diff
@@ -0,0 +1,54 @@
+package services.applications;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.Application;
+import play.libs.concurrent.HttpExecutionContext;
+import repository.ApplicationRepository;
+import services.program.ProgramDefinition;
+
+/** The service responsible for mediating access to the Application resource. */
+public final class ApplicationService {
+
+  private final ApplicationRepository applicationRepository;
+  private final HttpExecutionContext httpExecutionContext;
+
+  @Inject
+  ApplicationService(
+      ApplicationRepository applicationRepository, HttpExecutionContext httpExecutionContext) {
+    this.applicationRepository = checkNotNull(applicationRepository);
+    this.httpExecutionContext = checkNotNull(httpExecutionContext);
+  }
+
+  public Optional<Application> getApplication(
+      Optional<Application> application, ProgramDefinition program) {
+    if (application.isEmpty()
+        || !application
+            .get()
+            .getProgram()
+            .getProgramDefinition()
+            .adminName()
+            .equals(program.adminName())) return Optional.empty();
+    else return application;
```

Comment:
**suggestion (blocking):** Please review the Java style guide: https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used
```suggestion
    if (application.isEmpty()
        || !application
            .get()
            .getProgram()
            .getProgramDefinition()
            .adminName()
            .equals(program.adminName())) {
      return Optional.empty();
    } else {
      return application;
    }
```